src/main/java/se/sundsvall/casedata/service/util/mappers/PatchMapper.java

```diff
@@ -34,21 +34,25 @@ public final class PatchMapper {
 	private PatchMapper() {}
 
 	public static ErrandEntity patchErrand(final ErrandEntity errand, final PatchErrand patch) {
-		ofNullable(errand.getExtraParameters()).ifPresentOrElse(List::clear, () -> errand.setExtraParameters(new ArrayList<>()));
-		errand.getExtraParameters().addAll(toErrandParameterEntityList(patch.getExtraParameters(), errand));
-
-		Optional.ofNullable(patch.getCaseType()).ifPresent(caseType -> errand.setCaseType(caseType.name()));
-		Optional.ofNullable(patch.getExternalCaseId()).ifPresent(errand::setExternalCaseId);
-		Optional.ofNullable(patch.getPriority()).ifPresent(errand::setPriority);
-		Optional.ofNullable(patch.getDescription()).ifPresent(errand::setDescription);
-		Optional.ofNullable(patch.getCaseTitleAddition()).ifPresent(errand::setCaseTitleAddition);
-		Optional.ofNullable(patch.getDiaryNumber()).ifPresent(errand::setDiaryNumber);
-		Optional.ofNullable(patch.getPhase()).ifPresent(errand::setPhase);
-		Optional.ofNullable(patch.getStartDate()).ifPresent(errand::setStartDate);
-		Optional.ofNullable(patch.getEndDate()).ifPresent(errand::setEndDate);
-		Optional.ofNullable(patch.getApplicationReceived()).ifPresent(errand::setApplicationReceived);
-		Optional.ofNullable(patch.getFacilities()).ifPresent(facilities -> errand.getFacilities().addAll(patch.getFacilities().stream().map(facility -> toFacilityEntity(facility, errand.getMunicipalityId(), errand.getNamespace())).toList()));
-		Optional.ofNullable(patch.getSuspension()).ifPresent(
+
+		ofNullable(patch.getExtraParameters()).ifPresent(extraParameters ->
+		{
+			ofNullable(errand.getExtraParameters()).ifPresentOrElse(List::clear, () -> errand.setExtraParameters(new ArrayList<>()));
+			errand.getExtraParameters().addAll(toErrandParameterEntityList(extraParameters, errand));
+		});
```

Comment:
```
ofNullable(patch.getExtraParameters()).ifPresent(extraParameters -> errand.setExtraParameters(toErrandParameterEntityList(extraParameters);
```

This should be sufficient no? If the patch contains any previous extraparameters are either cleared or a new list is created. So we might aswell just set them directly.

Reply:
Cant be done like that due to hibernate, immutable lists and other nonsense.